flink-runtime/src/main/java/org/apache/flink/runtime/io/network/metrics/InputChannelMetrics.java

```diff
@@ -72,4 +72,47 @@ public Counter getNumBuffersInLocalCounter() {
 	public Counter getNumBuffersInRemoteCounter() {
 		return numBuffersInRemote;
 	}
+
+	private static class MultiCounterWrapper implements Counter {
+		private final Counter[] counters;
+
+		private MultiCounterWrapper(Counter ... counters) {
+			Preconditions.checkArgument(counters.length > 0);
+			this.counters = counters;
+		}
+
+		@Override
+		public void inc() {
+			for (Counter c : counters) {
+				c.inc();
+			}
+		}
+
+		@Override
+		public void inc(long n) {
+			for (Counter c : counters) {
+				c.inc(n);
+			}
+		}
+
+		@Override
+		public void dec() {
+			for (Counter c : counters) {
+				c.dec();
+			}
+		}
+
+		@Override
+		public void dec(long n) {
+			for (Counter c : counters) {
+				c.dec(n);
+			}
+		}
+
+		@Override
+		public long getCount() {
+			// assume that the counters are not accessed directly elsewhere
```

Comment:
I am not sure whether `getCounter` would be actually used. If used we should keep the return counter as previous structure. That means calling `new InputChannelMetrics(parentGroup, networkGroup) instead?